src/sql/workbench/services/commandLine/common/commandLineService.ts

```diff
@@ -136,11 +137,37 @@ export class CommandLineService implements ICommandLineProcessing {
 		profile.serverName = args.server;
 		profile.databaseName = args.database ? args.database : '';
 		profile.userName = args.user ? args.user : '';
-		profile.authenticationType = args.integrated ? 'Integrated' : 'SqlLogin';
+		profile.authenticationType = args.integrated ? Constants.integrated : args.aad ? Constants.azureMFA : (profile.userName.length > 0) ? Constants.sqlLogin : Constants.integrated;
 		profile.connectionName = '';
 		profile.setOptionValue('applicationName', Constants.applicationName);
 		profile.setOptionValue('databaseDisplayName', profile.databaseName);
 		profile.setOptionValue('groupId', profile.groupId);
-		return profile;
+		return this._connectionManagementService ? this.tryMatchSavedProfile(profile) : profile;
+	}
+
+	private tryMatchSavedProfile(profile: ConnectionProfile)
+	{
+		let match: ConnectionProfile = undefined;
+		// If we can find a saved mssql provider connection that matches the args, use it
+		let groups = this._connectionManagementService.getConnectionGroups([Constants.mssqlProviderName]);
+		if (groups && groups.length > 0)
+		{
+			let rootGroup = groups[0];
+			let connections = rootGroup.connections;
+			match = connections.find((c) => this.matchProfile(profile, c)) ;
+		}
+		return match ? match : profile;
+	}
+
+	// determines if the 2 profiles are a functional match
+	// profile1 is the profile generated from command line parameters
+	private matchProfile(profile1: ConnectionProfile, profile2: ConnectionProfile): boolean
```

Comment:
This would be useful to have on ConnectionProfile itself IMO

Reply:
this is a pretty specific definition of "match" that may not be useful for every consumer. Also there's a matches method already on connectionProfile that includes more properties. That one uses a case insensitive comparison on database name.

Reply:
Do we really want to have multiple definitions of "functional match"? It seems like that should be a constant thing. Having multiple definitions just makes it more likely we get it "wrong" somewhere and/or we have inconsistent behavior across the app based on what's doing the comparison. 

(The whole database case sensitivity is a separate issue too - that's a problem in a lot of our code and should be addressed more widely at some point)

Although looking at the one in connectionProfile I wonder if we should have two types of matches - one that just does core connection properties and one that does connection + UI properties (such as group ID and databaseDisplayName).

Reply:
i think this can stay as-is until there's some other need to reuse this particular type of comparison.